hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AndPredicate.java

```diff
@@ -99,6 +100,16 @@ public Predicate accept(Visitor visitor, Indexes indexes) {
         return new AndResultSet(smallestIndexedResult, otherIndexedResults, lsNoIndexPredicates);
     }
 
+    private int size(Set<QueryableEntry> result) {
+        // In case of AndResultSet and OrResultSet calling size() may be very expensive so quickerEstimated() size is used
+        if (result instanceof AndResultSet) {
```

Comment:
the question is whether to create an interface `WithSizeEstimator` or something. probably an overkill.